spec/classes/puppet_agent_spec.rb

```diff
@@ -52,7 +52,7 @@
         end
 
         it do
-          should contain_puppet__config__agent('server').with_value('foo.example.com')
+          should contain_puppet__config__main('server').with_value('foo.example.com')
```

Comment:
Well, the agent config inherits the main config class so there shouldn't be any need to duplicate the test. If the code being tested is in the puppet::config class then the test should be in the corresponding spec file for puppet::config.